src/main/java/Pat.java

```diff
@@ -0,0 +1,172 @@
+        try {
+            Task task = todoList.get(Integer.parseInt(arg) - 1);
+            task.markDone();
+            say("Nice! I've marked this task as done:");
+            say(task.getTask());
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            say("Please tell me a valid task to mark!");
+            say("mark [task number from list]");
+        }
+    }
+
+    private static void unmarkTask(String arg) {
+        try {
+            Task task = todoList.get(Integer.parseInt(arg) - 1);
+            task.markNotDone();
+            say("OK, I've marked this task as not done yet:");
+            say(task.getTask());
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            say("Please tell me a valid task to unmark!");
+            say("unmark [task number from list]");
+        }
+    }
+
+    private static void handleTodo(String arg) {
+        try {
+            String todo = arg.trim();
+            if (todo.isEmpty()) {
+                throw new IllegalArgumentException("Empty todo");
+            }
+            addToTodo(new Todo(todo));
+        } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+            say("Please give me a valid todo to add!");
+            say("todo [task]");
+        }
+    }
+
+    private static void handleDeadline(String arg) {
+        try {
+            String[] splitLineBy = arg.split(" /by ", 2);
+            String deadline = splitLineBy[0].trim();
+            String by = splitLineBy[1].trim();
+            if (deadline.isEmpty()) {
+                throw new IllegalArgumentException("Empty deadline");
+            } else if (by.isEmpty()) {
+                throw new IllegalArgumentException("Empty by");
+            }
+            addToTodo(new Deadline(deadline, by));
+        } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+            say("Please give me a valid deadline to add!");
+            say("deadline [task] /by [deadline]");
+        }
+    }
+
+    private static void handleEvent(String arg) {
+        try {
+            String[] splitLineFrom = arg.split(" /from ", 2);
+            String event = splitLineFrom[0].trim();
+            String[] splitLineTo = splitLineFrom[1].split(" /to ", 2);
+            String from = splitLineTo[0].trim();
+            String to = splitLineTo[1].trim();
+            if (event.isEmpty()) {
+                throw new IllegalArgumentException("Empty event");
+            } else if (from.isEmpty()) {
+                throw new IllegalArgumentException("Empty from");
+            } else if (to.isEmpty()) {
+                throw new IllegalArgumentException("Empty to");
+            }
+            addToTodo(new Event(event, from, to));
+        } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+            say("Please give me a valid event to add!");
+            say("event [task] /from [start] /to [end]");
+        }
+    }
```

Comment:
I like how you extracted these methods to improve your code quality. Good use of try and catch exceptions to handle invalid commands from the user.